server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -412,6 +421,10 @@ public DataStreamLifecycle getLifecycle() {
         return lifecycle;
     }
 
+    public DataStreamAutoShardingEvent getAutoShardingEvent() {
```

Comment:
Can we add a small blurb like 
```suggestion
    /**
     * Returns the latest auto sharding event that happened for this data stream
     */
    public DataStreamAutoShardingEvent getAutoShardingEvent() {
```
to clarify it's the *last* one, and not some amalgamation of all of them or something else